File: solutions/2026-06-29/solution.java
// Validate IP Address
// Difficulty: Medium
// URL: https://leetcode.com/problems/validate-ip-address/

public String validIPAddress(String s) {
  public boolean isValidPart(String part) {
    if (part.length() > 1 && part.startsWith("0")) return false;
    if (!part.matches("[0-9]+")) return false;
    return Integer.parseInt(part) <= 255 && Integer.parseInt(part) >= 0;
  }
  String[] parts = s.split(".");
  if (parts.length != 4) return "Invalid";
  for (String part : parts) {
    if (!isValidPart(part)) return "Invalid";
  }
  return "Valid";
}

File: solutions/2026-06-29/solution.py
# Validate IP Address
# Difficulty: Medium
# URL: https://leetcode.com/problems/validate-ip-address/

def validIPAddress(s: str) -> str: 
  def is_valid_part(part: str) -> bool: 
    if len(part) > 1 and part.startswith('0'): return False 
    if not part.isdigit(): return False 
    return 0 <= int(part) <= 255 
  parts = s.split('.') 
  if len(parts) != 4: return 'Invalid' 
  for part in parts: 
    if not is_valid_part(part): return 'Invalid' 
  return 'Valid'